packages/account/src/Sections/Verification/ProofOfIncome/proof-of-income-form.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { Field, Formik, Form, FormikErrors, FormikHelpers, FormikValues } from 'formik';
+import { AccountStatusResponse, DocumentUploadRequest } from '@deriv/api-types';
+import {
+    Autocomplete,
+    Button,
+    DesktopWrapper,
+    FormSubmitErrorMessage,
+    MobileWrapper,
+    SelectNative,
+} from '@deriv/components';
+import { useFileUploader } from '@deriv/hooks';
+import { localize, Localize } from '@deriv/translations';
+import { isEqualArray, WS } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import FilesDescription from 'Components/file-uploader-container/files-descriptions';
+import FormBody from 'Components/form-body';
+import FormFooter from 'Components/form-footer';
+import FormSubHeader from 'Components/form-sub-header';
+import { income_status_codes, poinc_documents_list } from 'Sections/Verification/ProofOfIncome/proof-of-income-utils';
+import FileUploaderContainer from '../../../Components/file-uploader-container';
+import { isServerError } from 'Helpers/utils';
+
+type TProofOfIncomeForm = {
+    onSubmit: (status: typeof income_status_codes[keyof typeof income_status_codes]) => void;
+};
+
+type TInitialValues = {
+    document_type: DocumentUploadRequest['document_type'] | '';
+};
+
+const ProofOfIncomeForm = observer(({ onSubmit }: TProofOfIncomeForm) => {
+    const [document_file, setDocumentFile] = React.useState<File[]>([]);
+    const [file_selection_error, setFileSelectionError] = React.useState<string | null>(null);
+
+    const { notifications, ui } = useStore();
+    const { addNotificationMessageByKey, removeNotificationMessage, removeNotificationByKey } = notifications;
+    const { is_mobile, is_desktop } = ui;
+
+    const { upload } = useFileUploader();
+
+    const initial_form_values: TInitialValues = {
+        document_type: '',
+    };
+
+    const validateFields = (values: TInitialValues) => {
+        const errors: FormikErrors<TInitialValues> = {};
+        const { document_type } = values;
+
+        if (!document_type || !poinc_documents_list.find(c => c.text === document_type)) {
```

Comment:
```suggestion
        if (!document_type || !poinc_documents_list.find(c => c.text === document_type)) {
```
could we somehow extract `!poinc_documents_list.find(c => c.text === document_type` into variable?